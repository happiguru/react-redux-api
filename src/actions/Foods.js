export const addFoods = (foods = []) => ({
  type: 'ADD_FOODS',
  foods,
});

export const updateFoods = (id = '', update = {}) => ({
  type: 'UPDATE_FOODS',
  id,
  update,
});
