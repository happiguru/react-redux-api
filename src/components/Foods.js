import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../styles/Food.css';

const Food = ({
  idMeal, strMealThumb, strMeal, strArea, strCategory,
}) => (
  <div className="col-xl-4 col-lg-3 col-md-6" key={idMeal}>
    <Link to={`/${idMeal}`}>
      <div className="single_recepie text-center">
        <div className="recepie_thumb">
          <img src={strMealThumb} alt={strMeal} />
        </div>
        <h3>{strMeal}</h3>
        <span>{strArea}</span>
        <p>{strCategory}</p>
      </div>
    </Link>
  </div>
);

Food.propTypes = {
  idMeal: PropTypes.string.isRequired,
  strMealThumb: PropTypes.string.isRequired,
  strMeal: PropTypes.string.isRequired,
  strArea: PropTypes.string.isRequired,
  strCategory: PropTypes.string.isRequired,
};

export default Food;
