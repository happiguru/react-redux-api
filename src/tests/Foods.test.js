import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Foods from '../components/Foods';
import TestStore from './store';
import foods from './data';

describe('Foods list', () => {
  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={TestStore}>
        <Router>
          <Foods foods={foods} />
        </Router>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
