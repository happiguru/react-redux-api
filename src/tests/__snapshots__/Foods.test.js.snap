// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foods list should match with snapshot 1`] = `[Function]`;

exports[`Renders Foods 1`] = `
<div
  className="row"
/>
`;
